pkg/adaptor/cloud/aws/types_test.go

```diff
@@ -43,3 +43,106 @@ func TestEmptyList(t *testing.T) {
 		t.Errorf("Expect 0 length, got %d", len(list))
 	}
 }
+
+func TestEmptyKeyValueFlag_Set(t *testing.T) {
+	// Empty KeyValueFlag will result in error
+	var flag keyValueFlag
+	err := flag.Set("")
+	if err == nil {
+		t.Errorf("Expect error, got nil")
+	}
+}
+
+func TestKeyValueFlag_Set(t *testing.T) {
+	tests := []struct {
+		// Add test name
+		name          string
+		input         string
+		expectedValue keyValueFlag
+		expectedError bool
+	}{
+		{
+			name:  "valid key value pair",
+			input: "key1=value1,key2=value2,key3=value3",
+			expectedValue: keyValueFlag{
+				"key1": "value1",
+				"key2": "value2",
+				"key3": "value3",
+			},
+			expectedError: false,
+		},
+		{
+			name:          "invalid key value pair",
+			input:         "invalid",
+			expectedValue: nil,
+			expectedError: true,
+		},
+		// Add more test cases "key1=value1, key2=value2" and "key1=value1,key2=value2, key3=value3"
+		// to cover all the cases
+		{
+			name:  "valid key value pair with spaces 1",
+			input: "key1=value1, key2=value2",
+			expectedValue: keyValueFlag{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			expectedError: false,
+		},
+		{
+			name:  "valid key value pair with spaces 2",
+			input: "key1=value1,key2=value2, key3=value3",
+			expectedValue: keyValueFlag{
+				"key1": "value1",
+				"key2": "value2",
+				"key3": "value3",
+			},
+			expectedError: false,
+		},
+		// Add test case for "key1=value1 key2=value2"
+		{
+			name:  "invalid key value pair separated with spaces",
+			input: "key1=value1 key2=value2",
+			expectedValue: keyValueFlag{
+				"key1": "value1 key2=value2",
+			},
+			expectedError: false,
```

Comment:
is it valid or should this fail? spaces are valid as value?

Reply:
yes, spaces are valid in value